src/init.cpp

```diff
@@ -1031,6 +1031,19 @@ bool AppInitParameterInteraction(const ArgsManager& args, bool use_syscall_sandb
         return InitError(_("No proxy server specified. Use -proxy=<ip> or -proxy=<ip:port>."));
     }
 
+    if (args.IsArgSet("-reindex-chainstate")) {
+        // indexes that must be deactivated to prevent index corruption, see #24630
+        if (args.GetBoolArg("-coinstatsindex", DEFAULT_COINSTATSINDEX)) {
+            return InitError(_("reindex-chainstate with active coinstatsindex is not possible. Please deactivate the index temporarily or perform a full -reindex, which will also rebuild the index."));
+        }
+        if (g_enabled_filter_types.count(BlockFilterType::BASIC)) {
+            return InitError(_("reindex-chainstate with active blockfilterindex is not possible. Please deactivate the index temporarily or perform a full -reindex, which will also rebuild the index."));
+        }
+        if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX)) {
+            return InitError(_("reindex-chainstate with active txindex is not possible. Please deactivate the index temporarily or perform a full -reindex, which will also rebuild the index."));
```

Comment:
In commit "init: disallow reindex-chainstate with optional indexes" (93fa8f651242af5354ba92f7aaec18e878d4f840)

Curious, but do you know how or whether txindex gets corrupted? #24630 only mentions the other two indexes. I guess even if the txindex doesn't get corrupted, it is probably still good to prevent `-reindex-chainstate` and `-txindex` from being used together if the current implementation is wastefully slow and inefficient, until we properly make the options more compatible.

Reply:
I don't think that it will get corrupted, because it doesn't have any kind of running variable like the other two indexes. However, it would get "rebuilt" by processing `BlockConnected()` signals for the entire chain (with the result being identical to the previously saved index data), which is probably not be what the user expects from running a `-reindex-chainstate` command. That's why I disabled it as well - I don't think it is strictly necessary and can drop this, if you / others would prefer that.